fuel-core-interfaces/src/txpool.rs

```diff
@@ -0,0 +1,127 @@
+use async_trait::async_trait;
+use fuel_tx::{Transaction, TxId};
+use std::sync::Arc;
+
+use fuel_tx::{ContractId, UtxoId};
+use thiserror::Error;
+
+use crate::db::{Error as DbStateError, KvStoreError};
+use fuel_storage::Storage;
+use fuel_tx::Bytes32;
+use fuel_vm::prelude::Contract;
+
+pub trait TxPoolDB:
+    Storage<Bytes32, Transaction, Error = KvStoreError>
+    + Storage<ContractId, Contract, Error = DbStateError>
+    + Send
+    + Sync
+{
+    fn transaction(&self, tx_hash: TxId) -> Option<Arc<Transaction>> {
+        Storage::<Bytes32, Transaction>::get(self, &tx_hash)
+            .ok()
+            .flatten()
+            .map(|t| Arc::new(t.as_ref().clone()))
+    }
+
+    fn contract_exist(&self, contract_id: ContractId) -> bool {
```

Comment:
In this context this is fine, in both cases with or without result i will behave in similar way.

Reply:
Hmm I guess it's true from the perspective of the mempool that if it can't find a contract, then it doesn't really matter _why_ it can't find it. You'd just discard the transaction. Still seems like a cleaner API to have it be a `Result` in case you want to be able to handle certain types of failures and e.g. retry a few times if it's a network or other transient failure. 🤷

Reply:
Generally speaking we'll likely want to be able to distinguish between erroneous data that someone has shared with us vs a failure to include in the tx pool due to local issues (i.e. db corruption). Otherwise building things like spam protection mechanisms will be much harder.

Reply:
This is very insignificant, in terms if database call fails, that is usually point of no return and as point of no return, the least of the problem of user is transaction not included, client will stop syncing.
But as this seems as last comment on pull request, and it is not big code change, i made a commit, and in John words "**L**et's **G**et **T**his **M**erged".